index.js

```diff
@@ -126,7 +126,9 @@ function replacePath (path, keys) {
       name: key || index++,
       delimiter: prefix || '/',
       optional: optional,
-      repeat: repeat
+      repeat: repeat,
+      offset: i + (prefix ? prefix.length : 0),
```

Comment:
Nice @blakeembrey! Sorry for the delay in checking this out been busy couple of days here. You think it's worth attaching keys _and_ tokens now? Since they're always computed anyway and will contain references to the same objects it will be cheapish. So in addition to `foo.keys` there is now also `foo.tokens`. In an ideal world you eventually deprecate `keys` since everything needed is in `tokens` anyway.